addons/website_sale/models/website.py

```diff
@@ -536,6 +536,8 @@ def _send_abandoned_cart_email(self):
         for website in self.search([]):
             if not website.send_abandoned_cart_email:
                 continue
+            if request:
+                request.website = website.with_context(request.context)
```

Comment:
seems surprising, request.website is usually only ever set when the request comes from the frontend (eg. from a controller called by the frontend)

now a new hack is added where the website is set when  we launch a cron for the case where we launch the cron manually (otherwise request would be falsy and we would have no issue)

IMO the issue is more on:

https://github.com/odoo/odoo/blob/dbc91f300a93832dee751e28fb0ef5d8f2855701/addons/website_sale/models/website.py#L471

that should have a check on website too, looking at the code, eg:

```py
request and hasattr(request, 'website')
# or
request and getattr(request, 'is_frontend', False)
```
if the code can be called from a cron manually (so have a request but not from frontend).